docs/settings api.md

```diff
@@ -0,0 +1,187 @@
+# Archipelago Settings API
+
+The settings API describes how to use installation-wide config and let the user configure them, like paths, etc. using
+host.yaml. For the player settings / player yamls see [options api.md](options api.md).
+
+The settings API replaces `Utils.get_options()` and `Utils.get_default_options()`
+as well as the predefined `host.yaml` in the repository.
+
+For backwards compatibility with APWorlds, some interfaces are kept for now and will produce a warning when being used.
+
+
+## Config File
+
+Settings use options.yaml (manual override), if that exists, or host.yaml (the default) otherwise.
+The files are searched for in the current working directory, if different from install directory, and in `user_path`,
+which either points to the installation directory, if writable, or to %home%/Archipelago otherwise.
+
+**Examples:**
+* C:\Program Data\Archipelago\options.yaml
+* C:\Program Data\Archipelago\host.yaml
+* path\to\code\repository\host.yaml
+* ~/Archipelago/host.yaml
+
+With the new settings API, AP can update the config file and can create it with default values and comments,
+if it does not exist.
+
+
+## Global Settings
+
+All non-world-specific settings are defined directly in settings.py.
+Each value needs to have a default. If the default should be `None`, define it as `typing.Optional` and assign `None`.
+
+To access a "global" config value, with correct typing, use one of
+```python
+from settings import get_settings, GeneralOptions, FolderPath
+from typing import cast
+
+x = get_settings().general_options.output_path
+y = cast(GeneralOptions, get_settings()["general_options"]).output_path
+z = cast(FolderPath, get_settings()["general_options"]["output_path"])
+```
+
+
+## World Settings
+
+Worlds can define the top level key to use by defining `settings_key: ClassVar[str]` in their World class.
+It defaults to `{folder_name}_options` if undefined, i.e. `worlds/factorio/...` defaults to `factorio_options`.
```

Comment:
comma after i.e.

Reply:
comma after i.e. (or e.g. for that matter) is usually optional and depend on which "fork" of English you use. They all agree one should precede it though, so I'm less concerned about this one.

Reply:
A quick search says that's a difference between American and British english. We had mostly BE in school, so not my fault :-D